### ISSUE TITLE
WCAG 3.2.2 violation on date input without confirmation

### ISSUE THREAD

**Issue by melad-m** [City-of-Helsinki/helsinki-design-system#1202]
## To which feature the request relates to
> What is the component/pattern/token/documentation your request is related to?

Date input without confirmation. 

## What
> What are you proposing? Be as precise as possible.

In the documentation it is mentioned:

> Date input without a confirmation
> Note! Since this variant closes the date picker immediately after selection, it violates [WCAG 3.2.2 On Input guideline](https://www.w3.org/TR/WCAG21/#on-input)

WCAG 3.2.2 states that a change in context include changes of:

> [user agent](https://www.w3.org/WAI/WCAG21/Understanding/on-input.html#dfn-user-agent);
> [viewport](https://www.w3.org/WAI/WCAG21/Understanding/on-input.html#dfn-viewport);
> focus;
> [content](https://www.w3.org/WAI/WCAG21/Understanding/on-input.html#dfn-content) that changes the meaning of the [Web page](https://www.w3.org/WAI/WCAG21/Understanding/on-input.html#dfn-web-page)

With an additional note:

> A change of content is not always a change of context. Changes in content, such as an expanding outline, dynamic menu, or a tab control do not necessarily change the context, unless they also change one of the above (e.g., focus).

A change in focus refer to focus being moved to a separate component, but in this case focus remains on the date input component after the popup closes. Thus not technically creating a violation of this criterion. 

I have consulted with other accessibility experts who do not regard this as a violation. Interested to hear your thoughts on this. 

## Why
> Why is this new feature needed? How is it better than the current version?

## Related material
> Optionally, you may attach screenshots, Sketch files etc. to make your request more understandable. Also, you may provide links to similar content in other design systems.

**Comment by laurakarhu** [comment]
Thank you for the feedback!

Back in the days (3 years ago?) when this was evaluated, we considered with the accessibility specialists that in this case it was better to follow strict interpretation of the criterion and warn component users that the solution might not be fully accessible. The user has no information that the menu will close directly after selecting a value and user has no control over it. This is an issue especially for screenreaders. If user has selected an incorrect value by accident (pressing selection when they aimed to still navigate further), the automatically closing date picker can be confusing.

But we are also aware that there are currently improved ways to navigate such content with screen readers and our implementation should be revisited. And when we do, this needs to be re-evaluated as well.
